cns/ipampool/monitor.go

```diff
@@ -334,10 +335,20 @@ func (pm *Monitor) decreasePoolSize(ctx context.Context, meta metaState, state i
 	tempNNCSpec.RequestedIPCount -= int64(len(pendingIPAddresses))
 	logger.Printf("[ipam-pool-monitor] Decreasing pool size, pool %+v, spec %+v", state, tempNNCSpec)
 
-	_, err := pm.nnccli.UpdateSpec(ctx, &tempNNCSpec)
-	if err != nil {
-		// caller will retry to update the CRD again
-		return errors.Wrap(err, "executing UpdateSpec with NNC client")
+	attempts := 0
+	if err := retry.Do(func() error {
+		attempts++
+		_, err := pm.nnccli.UpdateSpec(ctx, &tempNNCSpec)
+		if err != nil {
+			// caller will retry to update the CRD again
+			logger.Printf("failed to update NNC spec attempt #%d, err: %v", attempts, err)
+			return errors.Wrap(err, "executing UpdateSpec with NNC client")
+		}
+		logger.Printf("successfully updated NNC spec attempt #%d", attempts)
+		return nil
+	}, retry.Attempts(5), retry.DelayType(retry.BackOffDelay)); err != nil { //nolint:gomnd // ignore retry magic number
+		logger.Errorf("all attempts failed to update NNC during scale-down, state is corrupt: %v", err)
+		panic(err)
```

Comment:
does the retrier return the last err or something generic like "retrier reached max attempts"? Asking in case we should wrap this first

Reply:
it returns a new error of all of the error(s) encountered in the retry loop